karapace/kafka/producer.py

```diff
@@ -59,3 +62,41 @@ def partitions_for(self, topic: str) -> dict[int, PartitionMetadata]:
             return self.list_topics(topic).topics[topic].partitions
         except KafkaException as exc:
             raise_from_kafkaexception(exc)
+
+
+class AIOKafkaProducer:
+    """An async wrapper around `KafkaProducer` built on top of confluent-kafka.
+
+    Starting `AIOKafkaProducer` instantiates a `KafkaProducer` and starts a poll-thread.
+
+    The poll-thread continuously polls the underlying producer so buffered messages
+    are sent and asyncio futures returned by the `send` method can be awaited.
+    """
+
+    def __init__(
+        self,
+        bootstrap_servers: Iterable[str] | str,
+        loop: asyncio.AbstractEventLoop | None = None,
+        **params: Unpack[KafkaClientParams],
+    ) -> None:
+        self.loop = loop or asyncio.get_running_loop()
+        self.producer = KafkaProducer(bootstrap_servers, **params)
+
+        self.stopped = Event()
+        self.poll_thread = Thread(target=self.poll_loop)
+        self.poll_thread.start()
+
+    async def stop(self) -> None:
+        self.stopped.set()
+        self.poll_thread.join()
+
+    def poll_loop(self) -> None:
+        """Target of the poll-thread."""
+        while not self.stopped.is_set():
+            self.producer.poll(0.1)
```

Comment:
`poll` is blocking, that's why it needs its own thread.
In case there is messages to be sent it'll just keep doing that (equivalent behaviour to [calling `flush`](https://github.com/confluentinc/librdkafka/blob/e03d3bb91ed92a38f38d9806b8d8deffe78a1de5/src/rdkafka.c#L4379-L4387)). If there are no messages to be sent, this call will block for `0.1` second, making sleep superfluous.

Reply:
This would actually be a good comment to add to this line IMO :)

Reply:
> If there are no messages to be sent, this call will block for 0.1 second, making sleep superfluous.

yeah my naive question was more like: its this busy wait or its smart enough to block. But not that I think about it its fine, your your responsibility :)
I agree that the comment could be useful to have!